<launch>

    <arg name="solver" default="OpenSot"/>
    <arg name="prefix" default=""/>
    <arg name="use_goal_generator" default="false"/>
    <arg name="world_frame_link" default="l_sole"/>
    <arg name="gazebo" default="false"/>
    <arg name="logger" default="false"/>

    <env name="MATLOGGER_DISABLE_LOG" value="1"/>


    <param name="robot_description"
        textfile="$(find multi_contact_planning)/utils/urdf/cogimon.urdf"/>

    <param name="robot_description_semantic"
        textfile="$(find multi_contact_planning)/utils/srdf/cogimon.srdf"/>

    <param name="arc_description"
        textfile="$(find multi_contact_planning)/utils/urdf/arc.urdf"/>

    <param name="planner/problem_description_constraint"
        textfile="$(find cartesio_planning)/configs/problem/comanplus_constraint_stack.yaml"/>

    <param name="cartesian/problem_description"
        textfile="$(find cartesio_planning)/configs/problem/comanplus_goal_generation_stack.yaml"/>

    <param name="planner/problem_description_goal"
        textfile="$(find cartesio_planning)/configs/problem/comanplus_goal_generation_stack.yaml"/>


    <node pkg="cartesio_planning" type="planning_server" name="planning_server" output="screen">
<!--        <remap from="planner/start/joint_states" to="cartesian/solution"/>-->
        <param name="solver" value=""/>
        <param name="is_model_floating_base" value="true"/>
        <param name="use_goal_generator" value="$(arg use_goal_generator)"/>
        <param name="planner_config"
            textfile="$(find cartesio_planning)/configs/problem/comanplus_example_manipulation_config.yaml"/>
        <param name="world_frame_link" value="$(arg world_frame_link)"/>
        <param name="eps" value="5e-2"/>
    </node>

    <node pkg="cartesio_planning" type="planning_scene_publisher" name="planning_scene_publisher" output="screen"/>

    <node pkg="cartesio_planning" type="previewer" name="planner_previewer" output="screen">
	<param name="is_model_floating_base" value="true"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="world_connecter_planner_start"
        args="0 0 0 0 0 0 world planner/start/world 30"/>

    <node pkg="tf" type="static_transform_publisher" name="world_connecter_planner_goal"
        args="0 0 0 0 0 0 world planner/goal/world 30"/>

    <node pkg="tf" type="static_transform_publisher" name="world_connecter_planner_playback"
        args="0 0 0 0 0 0 world planner/world 30"/>

    <node pkg="tf" type="static_transform_publisher" name="world_connecter_planner_ci"
        args="0 0 0 0 0 0 world ci/world 30"/>

    <node pkg="tf" type="static_transform_publisher" name="world_connecter_planner_goal_sampler"
        args="0 0 0 0 0 0 world planner/goal_sampler/world 30"/>
    
    <node pkg="tf" type="static_transform_publisher" name="world_connecter_solution"
        args="0 0 0 0 0 0 world solution/world 30"/>
        
    <node pkg="tf" type="static_transform_publisher" name="world_connecter_NSPG"
        args="0 0 0 0 0 0 world NSPG/world 30"/>

    <node if="$(arg gazebo)" type="rviz" name="rviz" pkg="rviz"
        args="-d $(find cartesio_planning)/configs/rviz/example_comanplus_manipulation.rviz" output="screen"/>

    <node unless="$(arg gazebo)" type="rviz" name="rviz" pkg="rviz"
        args="-d $(find cartesio_planning)/configs/rviz/example_comanplus_manipulation_sim_piselloni.rviz" output="screen"/>
    
    <node if="$(arg use_goal_generator)" pkg="cartesian_interface" type="marker_spawner" name="goal_marker_spawner" output="screen">
        <param name="ns" value="planner/goal_sampler"/>
        <param name="tf_prefix" value="planner/goal_sampler"/>
    </node>

    <include file="$(find cartesio_planning)/launch/forza_giusta.launch" launch-prefix="bash -c 'sleep 5.0; $0 $@' ">
        <arg name="robot" value="false"/>
    </include>

    <group if="$(arg gazebo)">
        <node pkg="gazebo_ros" type="gazebo" name="gazebo"
            args="$(env ROBOTOLOGY_ROOT)/external/soap_bar_rrt/multi_contact_planning/examples/configs/world/cogimon_no_wall.world -namespace arc"/>

        <node pkg="gazebo_ros" type="spawn_model" name="arc_spawner" respawn="false" output="screen"
            args="-urdf -model arc -param arc_description"/>

        <node pkg="multi_contact_planning" name="xbot_mode_handler" type="xbot_mode_handler.py"
              output="screen" launch-prefix="bash -c 'sleep 5.0; $0 $@' "/>

        <include file="$(find cartesio_planning)/launch/forza_giusta.launch" launch-prefix="bash -c 'sleep 5.0; $0 $@' ">
            <arg name="robot" value="false"/>
        </include>

        <node pkg="cartesian_interface" name="force_estimation_node" type="force_estimation_node" launch-prefix="bash -c 'sleep 5.0; $0 $@' ">
            <param name="is_model_floating_base" value="true"/>
            <param name="model_type" value="RBDL"/>
            <param name="links" type="yaml" value="[l_ball_tip, r_ball_tip]"/>
            <param name="chains" type="yaml" value="[left_arm, right_arm]"/>
            <param name="l_ball_tip_dofs" type="yaml" value="[0,1,2]"/>
            <param name="r_ball_tip_dofs" type="yaml" value="[0,1,2]"/>
        </node>
    </group>

    <node if="$(arg logger)" pkg="multi_contact_planning" type="logger_node" name="logger_node" output="screen"
        args="/force_opt/force_r_sole/value /force_opt/force_r_sole/value
              /force_opt/force_l_sole/value /force_opt/force_l_sole/value
              /force_opt/force_r_ball_tip/value /force_opt/force_r_ball_tip/value
              /force_opt/force_l_ball_tip/value /force_opt/force_l_ball_tip/value
              /force_opt/force_r_sole/value /force_opt/force_r_sole/reference
              /force_opt/force_l_sole/value /force_opt/force_l_sole/reference
              /force_opt/force_r_ball_tip/value /force_opt/force_r_ball_tip/reference
              /force_opt/force_l_ball_tip/value /force_opt/force_l_ball_tip/reference
              /force_opt/pose/l_ball_tip
              /force_opt/pose/r_ball_tip
              /force_opt/pose/l_sole
              /force_opt/pose/r_sole
              /xbotcore/ft/l_leg_ft
              /xbotcore/ft/r_leg_ft
              /cartesian/force_estimation/l_ball_tip
              /cartesian/force_estimation/r_ball_tip"/>

<!-- mesh_viz for images-->
    <!--<node name="mesh_viz_floor" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.3"/>
        <param name="scale_y" type="double" value="0.05"/>
        <param name="scale_z" type="double" value="2.0"/>
        <rosparam param="position"> [0.8, 0.45, 0.5] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor"/>
        <param name="show_control_axis" type="bool" value="true"/>
        <remap from="mesh_viz_floor/floor" to="planner/collision_objects"/>
    </node>-->

<!--    <node name="mesh_viz_floor1" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="2.0"/>
        <param name="scale_y" type="double" value="50.0"/>
        <param name="scale_z" type="double" value="0.05"/>
        <rosparam param="position"> [0.0, 0.0, 0.0] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor1"/>
        <param name="show_control_axis" type="bool" value="true"/>
        <remap from="mesh_viz_floor1/floor1" to="planner/collision_objects"/>
    </node>-->

    <!-- mesh_viz for quadrupedal walking-->
    <node name="mesh_viz_floor" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.3"/>
        <param name="scale_y" type="double" value="0.05"/>
        <param name="scale_z" type="double" value="1.1"/>
        <rosparam param="position"> [0.76, 0.45, 0.55] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor/floor" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor1" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.3"/>
        <param name="scale_y" type="double" value="0.05"/>
        <param name="scale_z" type="double" value="1.1"/>
        <rosparam param="position"> [0.76, -0.65, 0.55] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor1"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor1/floor1" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor2" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.3"/>
        <param name="scale_y" type="double" value="1.15"/>
        <param name="scale_z" type="double" value="0.05"/>
        <rosparam param="position"> [0.76, -0.1, 1.125] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor2"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor2/floor2" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor3" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.05"/>
        <param name="scale_y" type="double" value="1.15"/>
        <param name="scale_z" type="double" value="2.0"/>
        <rosparam param="position"> [2.145, -0.1, 1.0] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor3"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor3/floor3" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor4" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="5.39"/>
        <param name="scale_y" type="double" value="1.3"/>
        <param name="scale_z" type="double" value="0.05"/>
        <rosparam param="position"> [-0.47, -0.1, -0.025] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor4"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor4/floor4" to="planner/collision_objects"/>
    </node>

    <!-- mesh_viz for wall_climbing-->
    <!--<node name="mesh_viz_floor" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="1.0"/>
        <param name="scale_y" type="double" value="1.5"/>
        <param name="scale_z" type="double" value="0.05"/>
        <rosparam param="position"> [0.5, 0., -0.025] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor/floor" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor1" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="1.0"/>
        <param name="scale_y" type="double" value="0.05"/>
        <param name="scale_z" type="double" value="3.8"/>
        <rosparam param="position"> [0.5, -0.73, 1.875] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor1"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor1/floor1" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor2" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="1.0"/>
        <param name="scale_y" type="double" value="0.05"/>
        <param name="scale_z" type="double" value="3.8"/>
        <rosparam param="position"> [0.5, 0.73, 1.875] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor2"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor2/floor2" to="planner/collision_objects"/>
    </node>-->

    <!-- mesh_viz for ladder -->
    <!--<node name="mesh_viz_floor" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.05, 0., 0.19] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor/floor" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor1" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.15, 0., 0.39] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor1"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor1/floor1" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor2" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.25, 0., 0.59] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor2"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor2/floor2" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor3" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.35, 0., 0.79] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor3"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor3/floor3" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor4" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.45, 0., 0.99] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor4"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor4/floor4" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor5" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.55, 0., 1.19] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor5"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor5/floor5" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor6" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.65, 0., 1.39] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor6"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor6/floor6" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor7" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.75, 0., 1.59] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor7"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor7/floor7" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor8" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.85, 0., 1.79] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor8"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor8/floor8" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor9" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="0.08"/>
        <param name="scale_y" type="double" value="1.0"/>
        <param name="scale_z" type="double" value="0.01"/>
        <rosparam param="position"> [1.95, 0., 1.99] </rosparam>
        <rosparam param="orientation"> [0.0, 0.0, 0.0, 1.0] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor9"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor9/floor9" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor10" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="2.24"/>
        <param name="scale_y" type="double" value="0.05"/>
        <param name="scale_z" type="double" value="0.1"/>
        <rosparam param="position"> [1.45, 0.525, 1.0] </rosparam>
        <rosparam param="orientation"> [0.0, -0.5256679, 0.0, 0.8506899] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor10"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor10/floor10" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor11" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="2.24"/>
        <param name="scale_y" type="double" value="0.05"/>
        <param name="scale_z" type="double" value="0.1"/>
        <rosparam param="position"> [1.45, -0.525, 1.0] </rosparam>
        <rosparam param="orientation"> [0.0, -0.5256679, 0.0, 0.8506899] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor11"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor11/floor11" to="planner/collision_objects"/>
    </node>
    <node name="mesh_viz_floor12" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="2.24"/>
        <param name="scale_y" type="double" value="0.05"/>
        <param name="scale_z" type="double" value="0.1"/>
        <rosparam param="position"> [2.35, 0.525, 1.0] </rosparam>
        <rosparam param="orientation"> [0.0, 0.5256679, 0.0, 0.8506899] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor12"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor12/floor12" to="planner/collision_objects"/>
    </node>

    <node name="mesh_viz_floor13" pkg="mesh_viz" type="mesh_viz_node" output="screen">
        <param name="mesh_path" value="$(find cartesio_planning)/utils/simple_box.stl" />
        <param name="scale_x" type="double" value="2.24"/>
        <param name="scale_y" type="double" value="0.05"/>
        <param name="scale_z" type="double" value="0.1"/>
        <rosparam param="position"> [2.35, -0.525, 1.0] </rosparam>
        <rosparam param="orientation"> [0.0, 0.5256679, 0.0, 0.8506899] </rosparam>
        <param name="frame_id" value="world"/>
        <param name="name_id" value="floor13"/>
        <param name="show_control_axis" type="bool" value="False"/>
        <remap from="mesh_viz_floor13/floor13" to="planner/collision_objects"/>
    </node>-->

</launch>
